### ISSUE TITLE
False negative from `vertical_parameter_alignment_on_call` when calling `.init`

### ISSUE THREAD

**Issue by CraigSiemens** [realm/SwiftLint#3185]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`vertical_parameter_alignment_on_call` will not flag incorrect indentations on calls to `.init(...)` when the type is being inferred from either the property it's being assigned to or the function it's being returned from.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'Example.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.39.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Home-brew
* Paste your configuration file:

```yml
opt_in_rules:
  - vertical_parameter_alignment_on_call
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? 11.4.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
struct Foo {
    let value1: Int
    let value2: Int
}

// This should trigger a violation
let falseNegativeProperty: Foo = .init(value1: 1,
                                 value2: 2)

// This should trigger a violation
func falseNegativeFunction() -> Foo {
    .init(value1: 1,
    value2: 2)
}

// This will trigger a violation if uncommented
// let trueNegative = Foo(value1: 1,
//                    value2: 2)
```

**Comment by CraigSiemens** [comment]
This is still an issue in `0.41.0`

**Comment by CraigSiemens** [comment]
> Please comment to prevent this issue from being closed. Thank you for your contributions!
🤷‍♂️ I did...
